### PR TITLE
BTS-1892 | Foxx generates invalid Swagger JSON `schema: true` if a response but no joi schema is specified

### PR DESCRIPTION
**Pull request by Simran-B**
Backport of #21233

### CODE REVIEW

File: js/server/modules/@arangodb/foxx/router/swagger-context.js [hunk 1]
```diff
@@ -655,6 +655,9 @@ function swaggerifyBody (schema, optional) {
 
 function joi2schema (schema) {
   if (!schema.isJoi) {
+    if (typeof schema !== "object") {
+      return;
+    }
     return schema;
   }
   return joiToJsonSchema(schema);
```
Comment (KVS85) [inline js/server/modules/@arangodb/foxx/router/swagger-context.js:663]:
```
  if (typeof schema !== "object") {
      return;
  else if (!schema.isJoi) { 
      return schema;
  } else {
      return joiToJsonSchema(schema);
```

Comment (KVS85) [reply js/server/modules/@arangodb/foxx/router/swagger-context.js:663]: Isn't a variable typo should precede a member call?

Comment (Simran-B) [reply js/server/modules/@arangodb/foxx/router/swagger-context.js:663]: I initially checked for `boolean` but extended it later. It doesn't matter if `schema.isJoi` is accessed first, JavaScript simply returns a value of `undefined` in this case.

I don't think the behavior would be any different with your version but it's JavaScript... On the other hand, the existing code doesn't cover certain corner cases anyway. For example, a more water-tight check would not use `typeof` but test for null identity, handle the possibility of an Object with a null prototype, and compare the object's prototype to that of the base Object prototype to cover e.g. `new Boolean()` which has the type `object` and not `boolean`.

Comment (palashkaria) [reply js/server/modules/@arangodb/foxx/router/swagger-context.js:663]: @Simran-B you don't need to do any of this manually. We have `lodash` as a dep, I think this might be the right place to use it

Comment (palashkaria) [reply js/server/modules/@arangodb/foxx/router/swagger-context.js:663]: even without lodash, I'd prefer moving the object check out, & doing it before `schema.isJoi` check. Otherwise this same question will be asked in the future

Comment (Simran-B) [reply js/server/modules/@arangodb/foxx/router/swagger-context.js:663]: I mean sure. But note that there's more code that only checks for the typeof string, missing the same corner cases, and it's not worth fixing as this is Foxx.
